loader/loader.go

```diff
@@ -968,11 +1003,15 @@ func fetchMatchedLiteral(ctx *tcontext.Context, router *router.Table, schema, ta
 func (l *Loader) restoreData(ctx context.Context) error {
 	begin := time.Now()
 
-	conn, err := createConn(l.cfg)
+	baseConn, err := l.toDB.GetBaseConn(ctx)
 	if err != nil {
 		return err
 	}
-	defer conn.baseConn.Close()
+	defer l.toDB.CloseBaseConn(baseConn)
+	dbConn := &DBConn{
```

Comment:
no `resetBaseConnFn` set for `dbConn`?

Reply:
okay, add `resetBaseConnFn ` to nil [here](https://github.com/pingcap/dm/pull/266/files#diff-b38e43f0fb016af3e8cecd2d12d779ecR1020), it's never use in this function

Reply:
> it's never use in this function

@3pointer is `resetBaseConnFn` may be used in `DBConn.executeSQL`? then panic when accessing `nil` pointer.

Reply:
You are right, it maybe met `resetBaseConnFn`, add [it](https://github.com/pingcap/dm/pull/266/files#diff-b38e43f0fb016af3e8cecd2d12d779ecR1018), because it's not a kind of worker connection, so it return error immediately

Reply:
OK, sounds strange, but we can refactor it later.